### PR TITLE
Add basic QSFP-DD/OSFP support for sfputil

### PR DESCRIPTION
**Pull request by zzhiyuan**
Enable decode of QSFP-DD/OSFP transceivers, and vendor information.

### CODE REVIEW

**Comment by zzhiyuan** [comment]
> Indentation is currently inconsistent. Please unify indentation to multiples of 4 spaces.

Fixed. Please check again and let me know if there are places I've missed. Thanks

File: sonic_sfp/inf8628.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        return sffbase.parse(self, self.vendor_rev, rev_raw_data, start_pos)
+
+    def parse_vendor_pn(self, pn_raw_data, start_pos):
+        return sffbase.parse(self, self.vendor_pn, pn_raw_data, start_pos)
+
+    def parse_vendor_sn(self, sn_raw_data, start_pos):
+        return sffbase.parse(self, self.vendor_sn, sn_raw_data, start_pos)
+
+    def dump_pretty(self):
+        if self.interface_data == None:
```
Comment (jleveque) [inline sonic_sfp/inf8628.py]: `if self.interface_data is None:`

File: sonic_sfp/inf8628.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            {'offset': 0,
+             'size': 16,
+             'type': 'str'}
+        }
+
+    def __init__(self, eeprom_raw_data=None):
+        self.interface_data = None
+        start_pos = 0
+
+        if eeprom_raw_data != None:
```
Comment (jleveque) [inline sonic_sfp/inf8628.py]: `if eeprom_raw_data is not None:`

File: sonic_sfp/inf8628.py (hunk 1 above)
Comment (zzhiyuan) [reply sonic_sfp/inf8628.py]: > if self.interface_data is None:

Done

File: sonic_sfp/inf8628.py (hunk 2 above)
Comment (zzhiyuan) [reply sonic_sfp/inf8628.py]: > if eeprom_raw_data is not None:

Done

**Comment by keboliu** [comment]
@zzhiyuan  would you please share the doc of  INF-8628? I need to implement an extension to get more information from the SFP eeprom, and since inf-8628 support was added so the extension should also cover it, however, I only can find a pdf doc containing the cover of in-8628 but without content, so want to know whether you can share.

**Comment by zzhiyuan** [comment]
> @zzhiyuan would you please share the doc of INF-8628? I need to implement an extension to get more information from the SFP eeprom, and since inf-8628 support was added so the extension should also cover it, however, I only can find a pdf doc containing the cover of in-8628 but without content, so want to know whether you can share.

http://www.qsfp-dd.com/wp-content/uploads/2018/09/QSFP-DD-mgmt-rev3p0-final-9-18-18-Default-clean.pdf

**Comment by keboliu** [comment]
@zzhiyuan thank you very much!

**Comment by keboliu** [comment]
@zzhiyuan Thanks for initiating the work for QSFP-DD/OSFP, want to understand whether you have a plan to implement the DOM parsing for QSFP-DD/OSFP? Also, want to get @lguohan and @jleveque 's view.

**Comment by zzhiyuan** [comment]
> @zzhiyuan Thanks for initiating the work for QSFP-DD/OSFP, want to understand whether you have a plan to implement the DOM parsing for QSFP-DD/OSFP? Also, want to get @lguohan and @jleveque 's view.

Currently, I don't have plan on implementing additional parsing. This PR was enough to recognize the type of transceiver and the vendor information. I am not sure about the scope of information needed for the rest of SONiC.

Thanks

**Comment by keboliu** [comment]
> > @zzhiyuan Thanks for initiating the work for QSFP-DD/OSFP, want to understand whether you have a plan to implement the DOM parsing for QSFP-DD/OSFP? Also, want to get @lguohan and @jleveque 's view.
> 
> Currently, I don't have plan on implementing additional parsing. This PR was enough to recognize the type of transceiver and the vendor information. I am not sure about the scope of information needed for the rest of SONiC.
> 
> Thanks

@zzhiyuan for the scope of DOM info needed for SONiC you can refer to this doc: https://github.com/Azure/SONiC/blob/master/doc/xrcvd/transceiver-monitor-hld.md#112-transceiver-dom-sensor-table

**Pull request merged**